### PR TITLE
Added status header for query upload requests

### LINKED ISSUE #5412: "FirebaseError: Firebase Storage: An unknown error occurred ... (storage/unknown)" when using resume()/resume()/cancel() Methods on resumable UploadTask while on "Demo" emulator
### [REQUIRED] Environment info

**firebase-tools:** 11.20.0

**Platform:** Windows

### [REQUIRED] Test case

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Firebase Storage Emulator Demo</title>
    <style>
      html {
        font-family: "Roboto", sans-serif;
        color: #999;
        background-color: #222;
      }
    </style>
  </head>
  <body>
    <h1>Firebase Storage Emulator Demo</h1>
    <p>
      This is a demo of the Firebase Storage Emulator. It is a simple web app
      that uploads files to Firebase Storage and lists the uploaded files.
    </p>
    <div>
      <input type="file" id="file" />
      <span id="state"></span>
    </div>
    <div>
      <button id="upload">Upload</button>
      <button id="pause">Pause</button>
      <button id="resume">Resume</button>
      <button id="cancel">Cancel</button>
    </div>

    <div id="files"></div>
  </body>
  <script type="module">
    import { initializeApp } from "https://www.gstatic.com/firebasejs/9.15.0/firebase-app.js";
    import {
      getStorage,
      ref,
      uploadBytesResumable,
      getDownloadURL,
      connectStorageEmulator,
    } from "https://www.gstatic.com/firebasejs/9.15.0/firebase-storage.js";

    // Firebase configuration
    const firebaseConfig = {
      projectId: "demo-ember",
      appId: "demo",
      apiKey: "demo",
      authDomain: "demo",
      storageBucket: "demo",
    };

    // Initialize Firebase
    initializeApp(firebaseConfig);

    // Use the Firebase Storage emulator
    connectStorageEmulator(getStorage(), "localhost", 9199);

    // Buttons
    const upload = document.getElementById("upload");
    const pause = document.getElementById("pause");
    const resume = document.getElementById("resume");
    const cancel = document.getElementById("cancel");

    const stateValue = document.getElementById("state");

    // Upload button click handler
    upload.addEventListener("click", (e) => {
      // Get the file
      const file = document.getElementById("file").files[0];

      // Create a storage ref
      const storageRef = ref(getStorage(), file.name);

      // Upload file and metadata to the object 'images/mountains.jpg'
      const uploadTask = uploadBytesResumable(storageRef, file);

      // Pause button click handler
      pause.addEventListener("click", (e) => {
        // console log that the user clicked the pause button, and make the log text amber
        console.log("%cUser clicked pause button", "color: orange");
        uploadTask.pause();
      });

      // Resume button click handler
      resume.addEventListener("click", (e) => {
        console.log("%cUser clicked resume button", "color: orange");
        uploadTask.resume();
      });

      // Cancel button click handler
      cancel.addEventListener("click", (e) => {
        console.log("%cUser clicked cancel button", "color: orange");
        uploadTask.cancel();
      });

      // Listen for state changes, errors, and completion of the upload.
      uploadTask.on(
        "state_changed",
        (snapshot) => {
          const progress =
            (snapshot.bytesTransferred / snapshot.totalBytes) * 100;
          console.log("Upload is " + progress + "% done");

          switch (snapshot.state) {
            case "paused":
              stateValue.innerHTML = "Paused";
              console.log("Upload is paused");
              break;
            case "running":
              stateValue.innerHTML = "Running";
              console.log("Upload is running");
              break;
          }
        },
        (error) => {
          stateValue.innerHTML = "ERROR";
          console.log(error);
        },
        () => {
          stateValue.innerHTML = "Complete";
          console.log("upload commplete");
        }
      );
    });
  </script>
</html>
```

### [REQUIRED] Steps to reproduce

Use the HTML file above to upload a file to the Emulator, then hit pause before it's finished.

Emulator is simply running from...
``firebase emulators:start --project demo-ember``

### [REQUIRED] Expected behavior

I'd expect the uploadTask to not return ``FirebaseError: Firebase Storage: An unknown error occurred, please check the error payload for server response. (storage/unknown)``.

### [REQUIRED] Actual behavior

[firebase-debug.log](https://github.com/firebase/firebase-tools/files/10393274/firebase-debug.log)
[ui-debug.log](https://github.com/firebase/firebase-tools/files/10393275/ui-debug.log)

### PR DESCRIPTION
**Pull request by maneesht**
Fixes #5412

### CODE REVIEW

File: scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts [hunk 1]
```diff
@@ -179,10 +193,21 @@ describe("Firebase Storage endpoint conformance tests", () => {
             "X-Goog-Upload-Command": "finalize",
           })
           .expect(200)
-          .then((res) => res.header["x-goog-upload-status"]);
+          .then((res) => res.header[uploadStatusHeader]);
 
         expect(uploadStatus).to.equal("final");
 
+        queryUploadStatus = await supertest(firebaseHost)
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts:200]: please create a new test, go/refactoring/test-one-thing-at-a-time

File: scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts [hunk 2]
```diff
@@ -390,6 +425,17 @@ describe("Firebase Storage endpoint conformance tests", () => {
           })
           .expect(200);
 
+        const queryUploadStatus = await supertest(firebaseHost)
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts:428]: please create a new test, go/refactoring/test-one-thing-at-a-time

File: scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts [hunk 3]
```diff
@@ -179,10 +193,21 @@ describe("Firebase Storage endpoint conformance tests", () => {
...
           })
           .expect(200)
-          .then((res) => res.header["x-goog-upload-status"]);
+          .then((res) => res.header[uploadStatusHeader]);
 
         expect(uploadStatus).to.equal("final");
 
+        queryUploadStatus = await supertest(firebaseHost)
+          .put(uploadURL.pathname + uploadURL.search)
+          // No Authorization required in finalize
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts:202]: remove

File: scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts [hunk 4]
```diff
@@ -162,6 +165,17 @@ describe("Firebase Storage endpoint conformance tests", () => {
           .expect(200)
           .then((res) => new URL(res.header["x-goog-upload-url"]));
 
+        let queryUploadStatus = await supertest(firebaseHost)
+          .put(uploadURL.pathname + uploadURL.search)
+          // No Authorization required in finalize
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts:170]: remove

File: scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts [hunk 5]
```diff
@@ -162,6 +165,17 @@ describe("Firebase Storage endpoint conformance tests", () => {
           .expect(200)
           .then((res) => new URL(res.header["x-goog-upload-url"]));
 
+        let queryUploadStatus = await supertest(firebaseHost)
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/conformance/firebase.endpoints.test.ts:168]: please create a new test, go/refactoring/test-one-thing-at-a-time

File: src/emulator/storage/server.ts [hunk 6]
```diff
@@ -44,8 +44,8 @@ export function createApp(
       exposedHeaders: [
         "content-type",
         "x-firebase-storage-version",
+        "X-Goog-Upload-Size-Received",
         "x-goog-upload-url",
-        "x-goog-upload-status",
```
Comment (tonyjhuang) [inline src/emulator/storage/server.ts:48]: why are we removing this?

**Review by tonyjhuang — CHANGES_REQUESTED** [review]

File: src/emulator/storage/server.ts (hunk 6 above)
Comment (maneesht) [reply src/emulator/storage/server.ts:48]: It's a duplicate :-)

**Review by tonyjhuang — APPROVED** [review]

**Pull request merged**